### ISSUE TITLE
Multiple issues while adding frameworks from SPM to framework

### ISSUE THREAD

**Issue by mkkroliks** [firebase/firebase-ios-sdk#12618]
### Description

Hello everyone! 

I created a [simple app](https://github.com/firebase/firebase-ios-sdk/files/14731192/testapp.zip) that contains two targets. One is the main app target - **TestApp**, and the second is called **TestAppServices**. Both of them use firebase-ios-sdk frameworks. My dependency manager is SPM. I encountered 3 different issues:

1. When **FirebaseCrashlytics** is added to **TestApp**  and **FirebaseCrashlytics** and **FirebaseAnalytics** is added to **TestAppServices**, the app compiles without any issues, but on the app launch the console prints the following log 

> objc[76509]: Class _TtC20FirebaseCoreInternalP33_F216F9F2B7BB01CB0DE5BF38ABBA23AD12BundleFinder is implemented in both /Users/maciej.krolikowski/Library/Developer/Xcode/DerivedData/TestApp-egjsmekdnwgwslbinrwkbaxbfnbw/Build/Products/Debug-iphonesimulator/PackageFrameworks/FirebaseCrashlytics_-F5B0BD00025CD43_PackageProduct.framework/FirebaseCrashlytics_-F5B0BD00025CD43_PackageProduct (0x1062e6050) and /Users/maciej.krolikowski/Library/Developer/Xcode/DerivedData/TestApp-egjsmekdnwgwslbinrwkbaxbfnbw/Build/Products/Debug-iphonesimulator/TestAppServices.framework/TestAppServices (0x10608c030). One of the two will be used. Which one is undefined.

2. When both FirebaseCrashlytics and FirebaseAnalytics are added to both targets, the app compiles with the following error 
![image](https://github.com/firebase/firebase-ios-sdk/assets/2151961/41cc4a2a-ccad-4fab-b0f3-16ec0d0cb622)

3. When **FirebaseCrashlytics** is added to **TestApp**  and **FirebaseCrashlytics** and **FirebaseAnalytics** is added to **TestAppServices**, and I import FirebaseCrashlytics, I'm getting the following error.
![image](https://github.com/firebase/firebase-ios-sdk/assets/2151961/b659836b-0f38-4abb-9649-8640065e4863)

I would be grateful for any advice on how to overcome these problems. 

[testapp.zip](https://github.com/firebase/firebase-ios-sdk/files/14731192/testapp.zip)

### Reproducing the issue

Install the attached app and follow the instruction from the description.

### Firebase SDK Version

10.23.0

### Xcode Version

15.3

### Installation Method

Swift Package Manager

### Firebase Product(s)

Analytics, Crashlytics

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by paulb777** [comment]
This is a duplicate of #11156. Firebase's binary distributions, including Firebase Analytics, are build as static xcframeworks and do not support being linked into dynamic frameworks. They can only be linked into the main app or other static frameworks.

**Comment by mkkroliks** [comment]
Thank you @paulb777! After changing mach-o type to a static library, it works perfectly without any logs. However, I still have a problem with issue nr 3 - `no such module FirebaseCrashlytics`. Can you help me one more time? 😃

**Comment by paulb777** [comment]
Reopening to investigate 3

**Comment by paulb777** [comment]
I'm reproducing 3 by adding `import FirebaseCrashlytics` to TestServicesAnalytics.swift and building. It does not matter if TestAppServices is dynamic or static.

![Screenshot 2024-03-29 at 9 17 08 AM](https://github.com/firebase/firebase-ios-sdk/assets/73870/3f35fcb5-c90b-4727-bfc9-faa33c68bef8)

**Comment by ncooke3** [comment]
Hi @mkkroliks, to fix #3, navigate to your framework target's build settings, and set **_Build libraries for distribution_** to **_No_**.

<img width="1095" alt="Screenshot 2024-03-29 at 1 37 15 PM" src="https://github.com/firebase/firebase-ios-sdk/assets/36927374/d72311f3-3133-4258-a284-9b75216d86d7">

SwiftPM source targets are not built with library evolution mode enabled, so by publicly importing them in a framework target with library evolution mode enabled, this error will appear.
